indico/modules/events/layout/models/menu.py

```diff
@@ -121,12 +129,43 @@ def default_data(self):
     def is_orphaned(self):
         return bool(self.name and self.default_data is None)
 
+    @property
+    def is_accessible(self):
+        from indico.modules.events.contributions import Contribution
+        from indico.modules.events.contributions.models.persons import ContributionPersonLink, SubContributionPersonLink
+        from indico.modules.events.contributions.models.subcontributions import SubContribution
+        from indico.modules.events.models.persons import EventPerson
+
+        if not self.event_ref.can_manage(session.user):
+            if (self.visibility == MenuEntryVisibility.registered_users
+                    and not self.event_ref.is_user_registered(session.user)):
+                return False
+            elif (self.visibility == MenuEntryVisibility.speakers
+                    and not EventPerson.query
```

Comment:
I'd really avoid putting a query in an if statement like this... better to add some util functions for this

Reply:
What's the best place to implement the util? Maybe a `is_speaker(self, user)` method in `EventPerson`?

Reply:
It's action on a user, not a person... I'd just put it as a local `_is_speaker(event, user)` method in this file (or util.py in the module, then without the underscore)